Classes/Decoder/UrlDecoder.php

```diff
@@ -1162,16 +1162,28 @@ protected function makeRealPhpArrayFromRequestVars(array $requestVariables) {
 	 * not set already.
 	 */
 	protected function mergeGetVarsFromDomainsConfiguration() {
-		$getVars = array();
 		// Convert the configuration into an $_GET-"friendly" format
-		// Maybe there's an nicer way to do this
-		$getVarsAsQueryString = http_build_query($this->configuration->getGetVarsToSet());
-		parse_str($getVarsAsQueryString, $getVars);
-
-		// Over-write $_GET-params to match earlier behavior …
-		ArrayUtility::mergeRecursiveWithOverrule($getVars, $_GET, TRUE, TRUE, FALSE);
-		// … and store it back
-		$_GET = $getVars;
+		$getVars = $this->makeRealPhpArrayFromRequestVars($this->configuration->getGetVarsToSet());
+
+		// Over-write with $_GET-params that $_GET-parmas have a "higher" priority
+		$GET = GeneralUtility::_GET();
+		if (!is_array($GET)) {
+			$GET = array();
+		}
+		ArrayUtility::mergeRecursiveWithOverrule($getVars, $GET, TRUE, TRUE, FALSE);
+
+		// Re-caclulate the cHash
```

Comment:
I understand your concerns and I'd like to remove this.
But how should TYPO3 display the proper content from a plug-in that needs a specific GET-param?
When the cHash isn't recalculated TYPO3 will display the the first cache entry (this happens for me at least) 😕 

I know, you probably won't answer, but is something wrong with my `_DOMAINS` configuration?
```php
'_DEFAULT' => array(
	...
	'pagePath' => array(
		'rootpage_id' => '1',
	),
	...
),
'_SUBDOMAINS' => array(
	...
	'pagePath' => array(
		'rootpage_id' => '42',
	),
	...
),
'_DOMAINS' => array(
	'decode' => array(
		'sub1.example.tld' => array(
			'GETvars' => array(
				'tx_myext_pi1[property]' => '1',
			),
			'useConfiguration' => '_SUBDOMAINS',
		),
		'sub2.example.tld' => array(
			'GETvars' => array(
				'tx_myext_pi1[property]' => '2',
			),
			'useConfiguration' => '_SUBDOMAINS',
		),
	),
	'encode' => array(
		array(
			'GETvar' => 'tx_myext_pi1[property]',
			'urlPrepend' => 'http://sub1.example.tld',
			'useConfiguration' => '_SUBDOMAINS',
			'value' => '1',
		),
		array(
			'GETvar' => 'tx_myext_pi1[property]',
			'urlPrepend' => 'http://sub2.example.tld',
			'useConfiguration' => '_SUBDOMAINS',
			'value' => '2',
		),
	),
),
```